### ISSUE TITLE
Handling of unbound TypeVars from external libraries 

### ISSUE THREAD

**Issue by Dzeri96** [microsoft/pyright#3446]
**Is your feature request related to a problem? Please describe.**
I'm using the library [RxPy](https://github.com/ReactiveX/RxPY)(v. 4.0.0) for some stream processing and have ran into an interesting case of unbounded `TypeVar`s, similar to [this issue](https://github.com/microsoft/pylance-release/issues/2253).

This is the signature of the function I'm calling:
```python
_T = TypeVar("_T")
_TKey = TypeVar("_TKey")
_TValue = TypeVar("_TValue")

def group_by(
    key_mapper: Mapper[_T, _TKey],
    element_mapper: Optional[Mapper[_T, _TValue]] = None,
    subject_mapper: Optional[Callable[[], Subject[_TValue]]] = None,
) -> Callable[[Observable[_T]], Observable[GroupedObservable[_TKey, _TValue]]]:
...
```
As you can see, the `element_mapper` and `subject_mapper` parameters are optional, making `_TValue` unbound. 

Using a result of the call to this function would make pyright report an error like this:

```
Type "_TValue@group_by" cannot be assigned to type "Dict[Unknown, Unknown]"
```
The type of the error falls under the `reportGeneralTypeIssues` category.

**Describe the solution you'd like**
Comparing my problem to the above-linked issue, the main difference is that I can't modify the used code easily.
Libraries aside, wouldn't it make sense to assume every unbound generics parameter as `Unknown` or `Any`? 
In cases like these, no type can be inferred, and the consumer of the function cannot do anything with the info the error provides.
If the signature of the function is invalid, a separate error should reside there in my opinion.

If this is not feasible, it would be nice to have a separate category for unbound `TypeVars` and allow the user to suppress them or handle the case explicitly.

**Comment by erictraut** [comment]
Thanks for the issue report. Could you do me a favor and provide a self-contained minimal code sample? The above sample references a bunch of symbols that are not imported and not part of stdlib. I presume they come from some submodule installed by RxPy, but I'm not familiar with that library.

**Comment by Dzeri96** [comment]
Here are two small examples from the same codebase.
Please install the library `rx` (v.4.0.0) and run the code in python3.8.

The first example produces exactly the error described in this issue.

```python
from typing import Any, Callable, Dict
from reactivex import GroupedObservable, Subject
import reactivex.operators as ops

sub = Subject()

group_func: Callable[[Dict[str, Any]], str] = lambda x: 'side_channel' if 'type' in x else 'main_channel'
res = sub.pipe(
    ops.group_by(group_func)
)

res.subscribe(
    lambda sub_channel: init_side_and_main_channel(sub_channel),
)

def init_side_and_main_channel(sub_ch: GroupedObservable[str, Dict]):
    pass
```

In the second example I remove the type hinting from `group_func`, which causes another very similar error to be displayed by pyright:
```python
from typing import Any, Dict
from reactivex import GroupedObservable, Subject
import reactivex.operators as ops

sub = Subject()

res = sub.pipe(
    ops.group_by(lambda x: 'side_channel' if 'type' in x else 'main_channel')
)

res.subscribe(
    lambda sub_channel: init_side_and_main_channel(sub_channel),
)

def init_side_and_main_channel(sub_ch: GroupedObservable[str, Dict]):
    pass
```

**Comment by erictraut** [comment]
Are you sure the library is `rx`? The latest published version of the library is 3.2. I think you perhaps meant `reactivex` instead?

**Comment by Dzeri96** [comment]
True, the name was changed and now it's reactivex.

**Comment by erictraut** [comment]
This will be addressed in the next release.

**Comment by Dzeri96** [comment]
Glad to hear it!

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.247, which I just published. It will also be included in the next release of pylance.
